source/ts/poll.ts

```diff
@@ -20,10 +23,26 @@ export async function pollUntil<T>(
           return;
         }
 
-        query();
+        queryTimer = setTimeout(query, 0);
       });
     };
 
     query();
   });
 }
+
+export async function waitForResultToBe<T>(
+  queryFunction: () => Promise<T>,
+  expectedResult: T,
+  timeoutMs = 10000
```

Comment:
Maybe this could be `DEFAULT_TIMEOUT`

Reply:
`DEFAULT_TIMEOUT` is defined in `app-connection` rather than this file (`poll`). As `pollUntil` and this function are just generic utilities it seems reasonable that they define their own timeouts? Or maybe we could define another for this file?

Reply:
I think it's okay to do one locally in this file.